src/io/MPIInputOutput.hpp

```diff
@@ -224,5 +226,62 @@ int readReduceValuesConsecutive(T* valuesStart, MPI_Offset numValues, const std:
   MPI_Info_free(&info);
   return readcount;
 }
+
+inline bool file_exists(const std::string& path){
+  struct stat buffer;
+  return(stat (path.c_str(), &buffer) == 0);
+}
+
+template <typename T>
+void writeStatsJSONfileRootOnly(const T* data, int sizeOfData, const std::string& path, MPI_Comm comm, bool replaceExistingFile = false) {
+
+    int mpi_size;
+    int mpi_rank;
+    MPI_Comm_rank(comm, &mpi_rank);
+    MPI_Comm_size(comm, &mpi_size);
+
+    std::vector<int> recvCounts;
+    std::vector<int> displacement;
+
+    if (mpi_rank == 0) {
+      recvCounts.resize(mpi_size);
+    }
+
+    MPI_Gather(&sizeOfData, 1, MPI_INT, recvCounts.data(), 1, MPI_INT, 0, comm);
+
+    std::string recvBuffer;
+
+    MPI_Barrier(comm);
+
+    int totLen=0;
+
+    if (mpi_rank == 0) {
+      displacement.resize(mpi_size);
+      displacement[0] = 0;
+
+      totLen += recvCounts[0];
+      for(int i=1; i<mpi_size; i++) {
+        totLen += recvCounts[i];
+        displacement[i] = displacement[i-1] + recvCounts[i-1];
```

Comment:
ah and this block here could be a `std::exclusive_scan` https://en.cppreference.com/w/cpp/algorithm/exclusive_scan